### ISSUE TITLE
converting cucumber.json to final_report.adoc

### ISSUE THREAD

**Issue by vikramvi** [rmpestano/cukedoctor#81]
@rmpestano 

I tried using sample code to convert cucumber.json to  final_report.adoc but getting compilation error

`@Test
    public void shouldSaveDocumentationIntoDisk(){
    	List<String> pathToCucumberJsonFiles = FileUtil.findJsonFiles("target/");
    	List<Feature> features = FeatureParser.parse(pathToCucumberJsonFiles);
    	DocumentAttributes attrs = DefaultConfig.getInstance().getDocumentAttributes;
    	attrs.toc("left").backend("html5")
    			.docType("book")
    			.icons("font").numbered(false)
    			.sourceHighlighter("coderay")
    		    .sectAnchors(true).sectLink(true);

    	CukedoctorConverter converter = Cukedoctor.instance(features, "Living Documentation", attrs);
    	converter.setFilename("/target/living_documentation.adoc");

    	converter.saveDocumentation();
    	assertThat(FileUtil.loadFile("/target/living_documentation.adoc")).exists();
    }`

which packages do I need to import for DefaultConfig ?

Getting error against Cukedoctor.instance  "The method instance(List<Feature>, DocumentAttributes) in the type Cukedoctor is not applicable for the arguments (List<Feature>, String, DocumentAttributes)"

can you please clarify ? Thanks in advance

Regards,
Vikram

**Comment by rmpestano** [comment]
Hi @vikramvi,

In fact, the samplein README is wrong, there is no **title** parameter in [instance method](https://github.com/rmpestano/cukedoctor/blob/master/cukedoctor-converter/src/main/java/com/github/cukedoctor/Cukedoctor.java#L25). Documentation title is part of [DocumentoAttributes](https://github.com/rmpestano/cukedoctor/blob/master/cukedoctor-converter/src/main/java/com/github/cukedoctor/api/DocumentAttributes.java#L10).

Just remove the "Living Documentation" parameter and you should be ok.

I'll update readme, thanks for the heads up.

**Comment by vikramvi** [comment]
@rmpestano  thanks for quick info.

Please clarify which package to import  for "DefaultConfig.getInstance().getDocumentAttributes;"

I'm getting error "DefaultConfig cannot be resolved"

Thanks,
Vikram

**Comment by rmpestano** [comment]
You can look [here for complete examples](https://github.com/rmpestano/cukedoctor/blob/master/cukedoctor-converter/src/test/java/com/github/cukedoctor/converter/CukedoctorConverterTest.java).

The example in question is this one:  https://github.com/rmpestano/cukedoctor/blob/master/cukedoctor-converter/src/test/java/com/github/cukedoctor/converter/CukedoctorConverterTest.java#L667.

Hope it helps

**Comment by vikramvi** [comment]
@rmpestano  2 updates on this 

Issue 1:

below needs to be updated in READ.ME
DocumentAttributes attrs = GlobalConfig.getInstance().getDocumentAttributes(); 

Issue 2:

converter.saveDocumentation(); throwing error as below

Nov 23, 2016 6:42:48 PM com.github.cukedoctor.util.FileUtil saveFile
SEVERE: Could not create file /target/living_documentation.adoc
java.io.IOException: No such file or directory
	at java.io.UnixFileSystem.createFileExclusively(Native Method)
	at java.io.File.createNewFile(File.java:1012)
	at com.github.cukedoctor.util.FileUtil.saveFile(FileUtil.java:132)
	at com.github.cukedoctor.converter.CukedoctorConverterImpl.saveDocumentation(CukedoctorConverterImpl.java:223)
	at com.serenity.appium.poc.CukeDoctor.shouldSaveDocumentationIntoDisk(CukeDoctor.java:35)

**Comment by rmpestano** [comment]
Hi again,

For issue 1: GlobalConfig is used by default if you don't pass attrs in Cukedoctor.getInstace, so if you don't want to change defaults it is not needed.

For issue 2: It's a bit strange because even the directory doesn't existed it should be created.

If you have a simple maven project which reproduces the issue to share then a can have a look at it.


Meanwhile, try to use just fileName instead of "/target/filename" to see if it works, it must create the documentation in the directory where Cukedoctor is being executed.

**Comment by vikramvi** [comment]
@rmpestano removing folder in method converter.setFilename has fixed the issue and I could see living_documentation.adoc got created under project's parent directory ?

Q 1) How to set directory under which one needs to get .adoc file created ?

Q2 ) section 6.2
       DocumentAttributes attrs = DefaultConfig.getInstance().getDocumentAttributes; 
       should be changed to 
       DocumentAttributes attrs = GlobalConfig.getInstance().getDocumentAttributes();

      I felt this was not clear in Issue 1: comment 

Please review

**Comment by rmpestano** [comment]
Hi @vikramvi,

Q1: It depends on which module you are using:

- _converter module_ use cukedoctorConverter.setFileName("path_to_directory") as you are already doing. I need to investigate why it is not creating directories when they don't exist.

- _maven plugin_ use **outputDir** attribute, see [example here](https://github.com/rmpestano/cukedoctor#72-example-of-configuration). 

- _main module (cli)_ use **-p** flag, see [example here](https://github.com/rmpestano/cukedoctor#811-command-line).

Q2: Will update readme, thanks again.

I hope it helps.

**Comment by vikramvi** [comment]
Thanks for fixes, kindly keep the issue open till setFilename issue is resolved.

as per readme, even when I mention existing folder name; script fails at converter.saveDocumentation().  Without passing foldername ( existing ) to setFilename all works well.

**Comment by rmpestano** [comment]
Hi @vikramvi,

have you tried to set filename without **leading slash**? e.g:

 ```
CukedoctorConverter converter = Cukedoctor.instance(features, "Living Documentation", attrs);
converter.setFilename("target/living_documentation.adoc");
```

leading slash means absolute path.

**Comment by vikramvi** [comment]
@rmpestano Thanks this worked.

Can you please fix this in 6.2 ?

Thanks,
Vikram

**Comment by rmpestano** [comment]
@vikramvi 

can we close this issue?
